pages/instructorAssessmentInstances/instructorAssessmentInstances.sql

```diff
@@ -52,3 +68,95 @@ INSERT INTO assessment_state_logs AS asl
     FROM
         results
 );
+
+-- BLOCK open_all
+WITH results AS (
+    UPDATE assessment_instances AS ai
+    SET
+        open = true,
+        date_limit = NULL,
+        auto_close = FALSE,
+        modified_at = now()
+    WHERE
+        ai.assessment_id = $assessment_id
+        AND ai.open = FALSE
+    RETURNING
+        ai.open,
+        ai.id AS assessment_instance_id
+)
+INSERT INTO assessment_state_logs AS asl
+        (open, assessment_instance_id, auth_user_id)
+(
+    SELECT
+        true, results.assessment_instance_id, $authn_user_id
+    FROM
+        results
+);
+
+-- BLOCK set_time_limit
+WITH results AS (
+    UPDATE
+        assessment_instances AS ai
+    SET
+        date_limit = CASE
+                     WHEN $base_time = 'null' THEN NULL
+                     ELSE GREATEST(current_timestamp,
+                                   (CASE
+                                    WHEN $base_time = 'start_date' THEN ai.date
+                                    WHEN $base_time = 'current_date' THEN current_timestamp
+                                    ELSE ai.date_limit
+                                    END) +
+                                   $time_add * INTERVAL '1 sec')
+                     END,
+        modified_at = now()
+    WHERE
+        ai.open
+        AND ai.id = $assessment_instance_id
+    RETURNING
+        ai.open,
+        ai.id AS assessment_instance_id,
+        ai.date_limit
+)
+INSERT INTO assessment_state_logs AS asl
+        (open, assessment_instance_id, date_limit, auth_user_id)
+(
+    SELECT
+        true, results.assessment_instance_id, results.date_limit, $authn_user_id
+    FROM
+        results
+);
+    
+-- BLOCK set_time_limit_all
+WITH results AS (
+    UPDATE
+        assessment_instances AS ai
+    SET
+        date_limit = CASE
+                     WHEN $base_time = 'null' THEN NULL
+                     ELSE GREATEST(current_timestamp,
+                                   (CASE
+                                    WHEN $base_time = 'start_date' THEN ai.date
+                                    WHEN $base_time = 'current_date' THEN current_timestamp
+                                    ELSE ai.date_limit
+                                    END) +
+                                   $time_add * INTERVAL '1 sec')
+                     END,
+        modified_at = now()
+    WHERE
+        ai.open
+        AND ai.assessment_id = $assessment_id
+        AND (ai.date_limit IS NOT NULL OR $base_time <> 'date_limit')
```

Comment:
Looks like Pascal! I much prefer C 😄
```suggestion
        AND (ai.date_limit IS NOT NULL OR $base_time != 'date_limit')
```